android/java/org/chromium/chrome/browser/search_engines/settings/BraveBaseSearchEngineAdapter.java

```diff
@@ -90,7 +91,8 @@ private static void applyChangesForYahooJp(List<TemplateUrl> templateUrls) {
 
     public static @SearchEngineAdapter.TemplateUrlSourceType int getSearchEngineSourceType(
             TemplateUrl templateUrl, TemplateUrl defaultSearchEngine) {
-        if (templateUrl.getIsPrepopulated()) {
+        if (templateUrl.getIsPrepopulated()
+                || CustomSearchEnginesUtil.isCustomSearchEngineAdded(templateUrl.getKeyword())) {
```

Comment:
why does this need `isCustomSearchEngineAdded` when it didn't before? This is also missing tests

Reply:
Currently, when a user adds a new search engine, it doesn't appear in the main "Search Engines" list right away because we filter out non-prepopulated search engines. It only becomes visible after the user sets it as the default.

We need save the search engine preferences upon adding. So when the user later visits the "Search Engines" page, we load any user-added search engines from the saved preferences and include them in the list.

We also need to use the same user saved search engine to populate the UI above the `Add a custom search engine` section.